ID-355 ID-350 Make typeahead more usable with NVDA

halfninja opened the pull request:
Declares the relationships between things better with ARIA, and updates a polite live region with what's going on. Tested in NVDA in Firefox.

halfninja committed:
ID-355 ID-350

halfninja committed:
ID-355 Desugar some template strings for compatibility.

lol768 commented:
I'd like to see the deploy preview, which I guess will happen when the ID7 dep version change is PR'd?

halfninja commented:
Yep - though I'll get it running locally so I can sort out any issues with this PR without having to release anything potentially duff to NPM, so I'll give you the link to that while it's running

halfninja committed:
ID-355 Move aria-label.

Didn't seem to honour it on the list item, needs to be on the containing item. Fixes crazy pronunciation when words are highlighted ('North C.A. Rolina')

halfninja committed:
ID-355 Fix labels for custom HTML items.

We have cases where we overide displayText and return HTML, which was going straight into HTML and being spoken out with fun results. Now will use itemTitle or a textified version of displayText.

halfninja committed:
ID-355 Fix words running together.

Just calling .text() can end up with words with no space between them. This adds a space between the boundary of common block-level elements.

halfninja commented:
With @lol768's assistance we've made sure it works tolerably in Android TalkBack, and made some changes to make the ARIA label be sensible when the custom displayText is actually HTML, so I declare this Good
